### ISSUE TITLE
function result is not used false positive on self

### ISSUE THREAD

**Issue by paveldvorak5** [kalessil/phpinspectionsea#1360]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 3.0.1 |
| Language level | PHP 7.3                    |

![image](https://user-images.githubusercontent.com/14863835/60851703-daae1280-a1e3-11e9-8475-bfbe274ac942.png)

```
class DateTime {
        /**
	 * @param int $count
	 *
	 * @return self|$this
	 */
	public function addSecond( int $count = null ): self {
		return $this->modify( ( $count ?? 1 ) . ' seconds' );
	}
}
```

`$currentDate->addSecond( self::CANDLE_INTERVAL );`

Current behaviour
--
Report that result is not used but report fluent is not checked
![image](https://user-images.githubusercontent.com/14863835/60851721-f87b7780-a1e3-11e9-9b89-533f55c25fa6.png)

When I remove ":self", the error disappears.

Expected behaviour
--
Do not report

Environment details
--
PhpStorm 2019.1.3
Build #PS-191.7479.51, built on June 18, 2019
JRE: 1.8.0_202-release-1483-b53 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.19.0-5-amd64

**Comment by kalessil** [comment]
Fixed!

**Comment by lsalazarm99** [comment]
It seems that the issue returned. Laravel uses many of these methods.

Here, `whereHas` and `whereIn` return `$this` (the first one passes through many methods, but end up returning `$this`).

![image](https://user-images.githubusercontent.com/42286051/123034471-26e1c580-d3af-11eb-9ba3-feef1c92be06.png)

Here is another widely used Laravel method with the same false positive.

![image](https://user-images.githubusercontent.com/42286051/123034618-77f1b980-d3af-11eb-8994-88cc8a52e42c.png)
